### PR TITLE
feat: add build time feature flags

### PR DESCRIPTION
**Pull request by keraf**
- [x] Add feature flags support (build time)
- [x] Write documentation

This PR is a draft but feel free to discuss this in the comments.

### CODE REVIEW

**Comment by szymonlesisz** [comment]
i'm not sure if top level package.json is the best place to declare it. Two reasons:
1. some of the projects are not strictly related to the suite (regardless the repo name) + there will be more in the future (connect, trezor-link, all low level stuff)
2. i can imagine situation where we want to disable something only for certain build (web or desktop) 

maybe it should be declared in each project separately? (suite-web, suite-desktop, suite-native)
or maybe to make life easier we should declare it in `suite/src/config` like we did with all other settings?
Pros:
- all possible configs/settings are in one place so everyone knows where to look
- you can use ts syntax => you have `typed const` to compare with somewhere in code where you want to use that flag (elimination of string typos or not existing values)

**Comment by prusnak** [comment]
> maybe it should be declared in each project separately?

I don't want to over-engineer stuff, but maybe we can have the flags both in the main `package.json` AND in the package-specific `packages/foo/package.json`. The final flags would be obtained by combining these two together (where package flags always overwrite the main flags, i.e. the "foo" flags have precedence over the "main" flags).

**Comment by szymonlesisz** [comment]
yeah, thats what leads me to the second thought to keep is as typescript config file to be able to use it like:
```
const FLAGS = {
    EXAMPLE_FLAG: true,
    EXAMPLE_FLAG_2: true,
};

const FLAGS_WEB = {
   ...FLAGS,
   EXAMPLE_FLAG: false,
}

const FLAGS_DESKTOP = {
   ...FLAGS,
   EXAMPLE_FLAG_2: false,
}

``` 

all in one place, pure and simple (and typed ;])

File: packages/suite/src/config/suite/features.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
+/**
+ * Feature flags configuration file
+ * @docs docs/misc/feature_flags.md
+ */
+
+// List of all feature flags and their explanation
+export type FeatureFlags =
```
Comment (szymonlesisz) [inline packages/suite/src/config/suite/features.ts]: this type can be generated directly from FLAGS object if you freeze it as const:
`type FeatureFlags = keyof typeof FLAGS;`

so you don't have to write them twice, in type and in object

File: packages/suite/src/config/suite/features.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+type FeatureFlagSets = {
+    [F in FeatureFlags]: boolean;
+};
+
+// General flags
+export const FLAGS: FeatureFlagSets = {
+    EXAMPLE: true,
+    EXAMPLE2: false,
+    // Add more flags here
+};
```
Comment (szymonlesisz) [inline packages/suite/src/config/suite/features.ts]: remove `FeatureFlagSets` type declaration and add `} as const;`

**Review by szymonlesisz — APPROVED** [review]

**Review by szymonlesisz — APPROVED** [review]
yeah!

**Pull request merged**